website/docs/about_code_generation.mdx

```diff
@@ -55,13 +79,132 @@ This includes but is not limited to:
 - some Riverpod features will be available only with code generation.
 
 At the same time, many applications already use code generation with packages such
-as [Freezed](https://pub.dev/packages/freezed) or [json_serializable](https://pub.dev/packages/json_serializable).  
+as [Freezed](https://pub.dev/packages/freezed) or [json_serializable](https://pub.dev/packages/json_serializable).
 In that case, your project probably is already setup for code generation, and
 using for Riverpod should be simple.
 
+## The Syntax
+
+### Defining a provider:
+When defining a provider using code generation, there are two key points to consider:
+1. <span style={RED_STYLE}>Sync/Async</span> initialization: Choose the appropriate return type based on whether the state is initialized
+synchronously or asynchronously.
+2. <span style={BLUE_STYLE}>Function/Class</span> syntax: Choose between function-based or class-based syntax depending on whether
```

Comment:
This description also feels a bit redundant.

I'd instead mention why use one vs the other. Such as saying how a function-based provider is pretty much the same as a class-based one without any method. 
And that the goal of the class-based syntax is to enable external objects to modify the state of a provider. As by default, providers don't have a way to be modified from outside.